File: Sem5/LP2/Assignment5.py
def book_tickets(movie, tickets, showtime):
    ticket_price = 100
    total_bill = tickets * ticket_price
    print(f"\n✅ Booking confirmed for {movie}!")
    print(f"Details: {tickets} tickets for the {showtime} show.")
    print(f"Total Amount: Rs. {total_bill}\n")

def main():
    movies = {
        "Interstellar": ["10:00 AM", "2:00 PM", "6:00 PM"],
        "The Dark Knight": ["11:00 AM", "3:00 PM", "7:00 PM"],
        "Dune": ["9:30 AM", "1:30 PM", "5:30 PM"]
    }

    print("Hello! Welcome to Movie Booking ChatBot.")
    print("You can ask for 'movies', 'showtimes', or just type a movie name to book.")

    while True:
        user_input = input("\nYou: ").strip() #remove leading and trailing whotespaces
        message = user_input.lower()

        # 1. Check if the user typed a movie name directly
        found_movie = next((m for m in movies if m.lower() in message), None)

        if "hi" in message or "hello" in message:
            print("ChatBot: Hello! Which movie are you looking for today?")

        elif "bye" in message:
            print("ChatBot: Goodbye! Enjoy your movie.")
            break

        # 2. If a movie name was found in the input, start booking immediately
        elif found_movie:
            print(f"ChatBot: I see you want to watch {found_movie}!")
            print(f"Available showtimes: {', '.join(movies[found_movie])}")
            
            showtime = input("ChatBot: Which showtime would you like? ")
            
            try:
                tickets = int(input("ChatBot: How many tickets do you need? "))
                book_tickets(found_movie, tickets, showtime)
            except ValueError:
                print("ChatBot: Invalid ticket number. Let's try again from the start.")

        elif "movie" in message:
            print("Currently playing:", ", ".join(movies.keys()))

        elif "showtime" in message or "timing" in message:
            for m, t in movies.items():
                print(f"{m}: {', '.join(t)}")
                
        elif "thankyou" in message or "Thank you" in message or "thank" in message:
            print("Your Welcome! It was pleasure helping you!!")

        else:
            print("ChatBot: I didn't catch that. Try typing a movie name like 'Dune' to book!")

if __name__ == "__main__":
    main()

File: Sem5/LP2/Assignment5.cpp
#include <iostream>
#include <string>
#include <map>
#include <vector>
#include <algorithm>

using namespace std;

// Convert string to lowercase
string toLower(string text) {
    transform(text.begin(), text.end(), text.begin(), ::tolower);
    return text;
}

// Display movies
void showMovies(map<string, vector<string>> &movies) {
    cout << "Here are the movies currently playing:\n";
    for (auto &movie : movies) {
        cout << "- " << movie.first << endl;
    }
}

// Display showtimes
void showTimings(map<string, vector<string>> &movies) {
    cout << "Here are the showtimes:\n";
    for (auto &movie : movies) {
        cout << movie.first << ":\n";
        for (auto &time : movie.second) {
            cout << "   " << time << endl;
        }
    }
}

// Booking function
void bookTickets(string movie, int tickets, string showtime) {
    int ticket_price = 100;
    int total_bill = tickets * ticket_price;

    cout << "\nBooking confirmed!\n";
    cout << "Movie: " << movie << endl;
    cout << "Tickets: " << tickets << endl;
    cout << "Showtime: " << showtime << endl;
    cout << "Total amount: Rs. " << total_bill << endl;
}

int main() {

    map<string, vector<string>> movies = {
        {"Interstellar", {"Morning: 10:00 AM", "Afternoon: 2:00 PM", "Evening: 6:00 PM"}},
        {"The Dark Knight", {"Morning: 11:00 AM", "Afternoon: 3:00 PM", "Evening: 7:00 PM"}},
        {"Dune", {"Morning: 9:30 AM", "Afternoon: 1:30 PM", "Evening: 5:30 PM"}}
    };

    string userInput;

    cout << "Hello! Welcome to Movie Booking ChatBot.\n";

    while (true) {
        cout << "\nYou: ";
        getline(cin, userInput);

        string message = toLower(userInput);

        if (message.find("hi") != string::npos || message.find("hello") != string::npos) {
            cout << "ChatBot: Hello! Ready to book some movie tickets?\n";
        }

        else if (message.find("movie") != string::npos) {
            showMovies(movies);
        }

        else if (message.find("showtime") != string::npos || message.find("timing") != string::npos) {
            showTimings(movies);
        }

        else if (message.find("book") != string::npos) {
            string movie, showtime;
            int tickets;

            cout << "Which movie would you like to watch? : ";
            getline(cin, movie);

            cout << "How many tickets do you need? : ";
            cin >> tickets;
            cin.ignore();  // clear buffer

            cout << "At what time would you like to watch the movie? : ";
            getline(cin, showtime);

            bookTickets(movie, tickets, showtime);
        }

        else if (message.find("bye") != string::npos || message.find("goodbye") != string::npos) {
            cout << "ChatBot: Thank you! Enjoy your movie. Goodbye!\n";
            break;
        }

        else {
            cout << "ChatBot: Sorry, I didn't understand. You can ask about movies, showtimes, or book tickets.\n";
        }
    }

    return 0;
}